### PR TITLE
fix(models): remove unused model

### PR DESCRIPTION
**Pull request by mars-lan**
Fixes https://github.com/linkedin/datahub/issues/1719

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by mars-lan** [comment]
Actually this model is still being referenced: https://github.com/linkedin/datahub/blob/1283dd3ff49ea169ff5b8b24eec7952bd3f70e09/gms/api/src/main/pegasus/com/linkedin/dataset/Dataset.pdl#L67. Let's do a proper refactoring in a follow-up PR.

File: gms/api/src/main/pegasus/com/linkedin/dataset/Dataset.pdl [hunk 1]
```diff
@@ -61,11 +61,6 @@ record Dataset includes DatasetKey, ChangeAuditStamps, VersionTag {
    */
   uri: optional Uri
 
-  /**
-   * List of dataset instances for this dataset
-   */
-  deploymentInfos: array[DeploymentInfo] = [ ]
```
Comment (jplaisted) [inline gms/api/src/main/pegasus/com/linkedin/dataset/Dataset.pdl:67]: Is this backwards incompatible?

Comment (mars-lan) [reply gms/api/src/main/pegasus/com/linkedin/dataset/Dataset.pdl:67]: It is, luckily no one is producing nor consuming the field so should be fine.

Comment (jplaisted) [reply gms/api/src/main/pegasus/com/linkedin/dataset/Dataset.pdl:67]: Are we sure about that? I'll approve this, but it seems like we maybe need a better stand around breaking changes.

Maybe we should add an experimental branch or something? Then we reserve the right to break it whenever we want. :p 

Granted, might be hard to manage an experimental branch and keep it in sync with master. idk?

Comment (mars-lan) [reply gms/api/src/main/pegasus/com/linkedin/dataset/Dataset.pdl:67]: There's a reason why we're on 0.x :)

**Pull request merged**